docs(agoric-cli): add agoric run; deprecate beta features (#904)
Agoric/documentation: main/guides/agoric-cli/README.md

  install [force-sdk-version]                    install Dapp dependencies
  follow [options] <path-spec...>                follow an Agoric Casting leader
  run <script> [script-args...]                  run a script with all your user privileges and some Agoric endowments
  deploy [options] [script...]                   run multiple scripts with all your user privileges against the local Agoric
                                                 VM
  publish [options] [bundle...]                  publish a bundle to a Cosmos chain
  wallet [options]                               wallet commands
  start [options] [profile] [args...]            start an Agoric VM
  help [command]                                 display help for command
```

## agoric run

Run a script with all your user privileges and some Agoric endowments.

**Usage**:

```sh
agoric run [options] <script> [script-args...]
```

See also:

- [`agoric run` supports bundling proposal modules with writeCoreProposal](https://github.com/Agoric/agoric-sdk/discussions/8087#discussioncomment-6534378) discussion [#8087](https://github.com/Agoric/agoric-sdk/discussions/8087) 2023-07

## ~~agoric init~~

> _Deprecated in favor of [yarn create @agoric/dapp](../getting-started/#creating-your-dapp-from-a-template)_

Reviewer: I don't know the right verbiage, but it may be worthwhile to say something like:
```suggestion
_Internal implementation for the preferred [yarn create @agoric/dapp](../getting-started/#creating-your-dapp-from-a-template)_
```
if we removed `agoric init`, then the `yarn create @agoric/dapp` form would stop working.

- **--dapp-base &lt;base-url>**: The location where the dapp template can be found. If this argument isn't passed, then the default value of `git://github.com/Agoric/` is used.
- **-V**, **--version**: Outputs the version number.
- **--docker-tag &lt;tag>**: Uses the specified tag of any started Docker containers. Defaults to `latest`.
- **--sdk**: Uses the Agoric SDK containing this program.
- **-v**, **--verbose**: Causes the command to be run in verbose mode.
- **-h**, **--help**: Displays help for the command.

**Examples**:

```
agoric init demo
```

Creates a directory named "demo" with contents copied from the default dapp-template _dapp-fungible-faucet_.

```
agoric init my-first-contract --dapp-template dapp-simple-exchange
```

Creates a directory named _my-first-contract_ with contents copied from the dapp-template _dapp-simple-exchange_.

```
agoric init my-contract --dapp-template dapp-skeleton --dapp-base file:///home/contracts
```

Creates a directory named _my-contract_ with contents copied from a dapp-template named _dapp-skeleton_ located under the URL _file:///home/contracts_.

## ~~agoric install~~

> _Deprecated_

Author: > `agoric install` can still be used to change the NPM versions of the sdk packages.

I'm not clear on what that means. Care to sketch a sentence or two of documentation?

We don't have a test for this feature, by chance, do we?

> _Deprecated_

Reviewer 2: `agoric install <npmtag>` is definitely tested and used in tests. It updates all the known agoric dependencies to that npm tag. However I think the functionality is currently overloaded, and it's likely acceptable to not advertise it until we have a more complete story, which also handles the issues of version tearing we've noticed with endo packages.
